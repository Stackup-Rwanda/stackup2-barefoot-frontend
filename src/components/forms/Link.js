import React from 'react';
import Link from '@material-ui/core/Link';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';

const useStyles = makeStyles({
  link: {
    textAlign: 'center',
    margin: '5%',
  },
});
const Links = ({ children }) => {
  const classes = useStyles();
  return (
    <div className={classes.link}>
      <Link href="/" variant="body2" style={{ color: 'white', fontSize: '14px' }}>
        {children}
      </Link>
    </div>
  );
};
Links.propTypes = {
  children: PropTypes.string.isRequired,
};
export default Links;
