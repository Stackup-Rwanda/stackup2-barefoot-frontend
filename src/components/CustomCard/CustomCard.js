import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Link from '@material-ui/core/Link';
import Paper from '@material-ui/core/Paper';
import Remove from '@material-ui/icons/Remove';
import PropTypes, { string } from 'prop-types';

const useStyles = makeStyles((theme) => ({
  icon: {
    color: theme.palette.primary.dark,
  },
}));

const CustomCard = ({ title, items }) => {
  const classes = useStyles();

  return (
    <Paper elevation={3} className="card">
      <div className="card-title">{title}</div>
      <div className="testimony-bar mb24">
        <Remove className={classes.icon} size={32} />
      </div>
      <div className="card-items">
        {
          items.map((item, index) => <Link className="card-item" href="/" key={index}>{item}</Link>)
        }
      </div>
    </Paper>
  );
};

CustomCard.propTypes = {
  title: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(string).isRequired,
};

export default CustomCard;
