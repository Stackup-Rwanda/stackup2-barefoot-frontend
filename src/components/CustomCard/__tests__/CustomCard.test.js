/* eslint-disable no-undef */
import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import Link from '@material-ui/core/Link';
import { cleanup } from '@testing-library/react';
import CustomCard from '../CustomCard';

const title = 'Our Company';
const items = ['Careers', 'About', 'Vision'];

describe('<CustomCard />', () => {
  afterEach(() => cleanup);

  it('Should match the CustomCard snapshot', () => {
    const component = createMount()(
      <CustomCard title={title} items={items} />,
    );

    expect(component.html).toMatchSnapshot();
    expect(component.find(Link)).toHaveLength(3);
  });
});
