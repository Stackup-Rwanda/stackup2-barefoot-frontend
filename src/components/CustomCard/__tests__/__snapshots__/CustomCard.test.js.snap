// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomCard /> Should match the CustomCard snapshot 1`] = `[Function]`;
